In [50]:
import pandas as pd
import numpy as np
from sklearn.model_selection import train_test_split
from sklearn.linear_model import LogisticRegression
from sklearn.metrics import accuracy_score, confusion_matrix

In [51]:
df = pd.read_csv('payment_fraud.csv', header=0)
df.head()

accountAgeDays  numItems  localTime paymentMethod  paymentMethodAgeDays  \
0              29         1   4.745402        paypal             28.204861   
1             725         1   4.742303   storecredit              0.000000   
2             845         1   4.921318    creditcard              0.000000   
3             503         1   4.886641    creditcard              0.000000   
4            2000         1   5.040929    creditcard              0.000000   

   label  
0      0  
1      0  
2      0  
3      0  
4      0

In [52]:
df.dtypes

accountAgeDays            int64
numItems                  int64
localTime               float64
paymentMethod            object
paymentMethodAgeDays    float64
label                     int64
dtype: object

In [53]:
df['paymentMethod'].unique()

array(['paypal', 'storecredit', 'creditcard'], dtype=object)

In [56]:
#encode [paymentMethod] column
from sklearn.preprocessing import LabelEncoder
df['paymentMethod'] = df['paymentMethod'].astype('str')
df[['paymentMethod']]=df[['paymentMethod']].apply(LabelEncoder().fit_transform)

<p> we select Logestic Regression for this problem as the we targeting binary classification (fraud or not fraud)</p>

In [58]:
#train test split
X_train, X_test, y_train, y_test = train_test_split(
    df.drop('label', axis=1), df['label'],
    test_size=0.33, random_state=17)

In [59]:
y_train

13272    0
22716    0
30255    0
7676     0
15105    0
        ..
31044    0
25631    0
33174    0
34959    0
10863    0
Name: label, Length: 26278, dtype: int64

In [60]:
clf = LogisticRegression()

clf.fit(X_train, y_train)
# Make predictions on test set
y_pred = clf.predict(X_test)
from sklearn.metrics import accuracy_score
print(accuracy_score(y_pred, y_test)) 




1.0


<p>Wow ,model gave an accuracy of 100 per cent by using the logistic regression algorithm</p>

<h3> model evaluation</h3>

<p>usually evaluation done with x_test and y_pred</p>

In [61]:
print(confusion_matrix(y_test, y_pred))

[[12753     0]
 [    0   190]]


<p>190 of transactions are fraudulent and 12753 are not fraudulant</p>